### PR TITLE
update test_setup_sregs

### PR DESCRIPTION
**Pull request by serban300**
Issue #, if available: #1016 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: arch/Cargo.toml [hunk 1]
```diff
@@ -14,3 +14,4 @@ memory_model = { path = "../memory_model" }
 
 [dev-dependencies]
 rand = ">=0.5.5"
+cpuid = { path = "../cpuid" }
```
Comment (andreeaflorescu) [inline arch/Cargo.toml]: I find it a bit strange for the arch crate to depend on `cpuid`. I will take a look at the code to see if maybe we can find another way of testing setup_sregs.

Comment (serban300) [reply arch/Cargo.toml]: I could remove the `if` based on the cpu vendor and just do 
```
sregs.gs.g = 1;
```
all the time in `test_setup_sregs`

What do you think ?

Comment (andreeaflorescu) [reply arch/Cargo.toml]: Sounds good for the moment. We can re-visit the decision in the future, but right now I don't see any disadvantage in doing just that.

Comment (serban300) [reply arch/Cargo.toml]: Ok. Done.

File: cpuid/src/transformer/amd.rs [hunk 2]
```diff
@@ -60,8 +60,6 @@ mod test {
 
     #[test]
     fn test_update_extended_cache_topology_entry() {
-        use cpu_leaf::leaf_0x8000001d::*;
```
Comment (andreeaflorescu) [inline cpuid/src/transformer/amd.rs]: This was already fixed by a different commit.

**Pull request merged**